bleak/backends/dotnet/client.py

```diff
@@ -168,79 +170,117 @@ async def connect(self, **kwargs) -> bool:
             return_type=BluetoothLEDevice,
         )
 
+        # Called on disconnect event or on failure to connect.
+        def handle_disconnect():
+            if self._connection_status_changed_token:
+                self._requester.remove_ConnectionStatusChanged(
+                    self._connection_status_changed_token
+                )
+                self._connection_status_changed_token = None
+
+            if self._requester:
```

Comment:
Here we (potentially) dispose of the requester before the session, which is opposite from how it is done in `disconnect`. Will that be a problem do you think?